gto-support-db-coroutines/src/main/kotlin/org/ccci/gto/android/common/db/CoroutinesAsyncDao.kt

```diff
@@ -0,0 +1,24 @@
+package org.ccci.gto.android.common.db
+
+import android.database.sqlite.SQLiteDatabase
+import kotlinx.coroutines.async
+
+interface CoroutinesAsyncDao : CoroutinesDao {
+    fun <T : Any> findAsync(clazz: Class<T>, vararg key: Any) = coroutinesScope.async { find(clazz, *key) }
+    fun <T : Any> getAsync(query: Query<T>) = coroutinesScope.async { get(query) }
+    fun getCursorAsync(query: Query<*>) = coroutinesScope.async { getCursor(query) }
+
+    fun insertAsync(obj: Any) = insertAsync(obj, SQLiteDatabase.CONFLICT_NONE)
+    fun insertAsync(obj: Any, conflictAlgorithm: Int) = coroutinesScope.async { insert(obj, conflictAlgorithm) }
+
+    fun updateAsync(obj: Any) = updateAsync(obj, *getFullProjection(obj.javaClass))
```

Comment:
I think I'm a bit confused about why these three functions have the same name. Just different use cases?

Reply:
the extra methods is because of it mirroring the underlying `Dao` API, but providing Async varieties of the different methods. I'm going to take a quick cleanup pass to try and utilize default params which may get rid of a couple of these.

Reply:
I can't consolidate these because you can't use a default value with a vararg parameter, this prevents the first and second one from being consolidated.

The 3rd one is a batch update method that will update fields on multiple objects matching the `where` expression at once.